src/test/java/com/training/api/tests/shorex/ShorexTests.java

```diff
@@ -64,4 +68,34 @@ public void testValidateProductTypeName() {
                     .isEqualTo("Shore Excursion");
         }
     }
+
+    @Test
+    public void testValidateOfferingDate() throws ParseException {
+
+        String pattern = "yyyyMMdd";
+        String offeringDate = "";
+        Date searchDate = new SimpleDateFormat(pattern)
+                .parse(map.get("shorexUrl").substring(map.get("shorexUrl").indexOf("?"))
+                        .replaceAll("\\D+", ""));
+
+        com.training.pojos.shorex.validate.Response responseElement =
+                new RestEngine().getResponse(map.get("shorexUrl"), headerMap).as(com.training.pojos.shorex.validate.Response.class);
+
+        Assertions.assertEquals(200, responseElement.getStatus(), "REQUEST UNSUCCESSFUL");
+
+        for (int productIndex = 0; productIndex < responseElement.getPayload().getProducts().size(); productIndex++) {
+            for (int offeringIndex = 0; offeringIndex < responseElement.getPayload()
+                    .getProducts().get(productIndex)
+                    .getOffering().size(); offeringIndex++) {
+
+                offeringDate = responseElement.getPayload()
+                                .getProducts().get(productIndex)
+                                .getOffering().get(offeringIndex).getOfferingDate();
+
+                softAssertions.assertThat(new SimpleDateFormat(pattern).parse(offeringDate))
+                        .as("Sail date is before Search Date")
+                        .isAfterOrEqualsTo(searchDate);
```

Comment:
This looks to be a little complex in readability, so lets, maybe before the two for loops store the products in a list by using this : 
`List<Products> products=responseElement.getPayload().getProducts()`
and then use this list in the for loops.